scripts/plot_dataframe.py

```diff
@@ -131,52 +151,68 @@ def plot_analyzed_dataframe(args: Namespace) -> None:
+    """
+
     tick_interval = strategy.ticker_interval
+    timerange = Arguments.parse_timerange(args.timerange)
 
-    # Load pair tickers
     tickers = {}
     if args.live:
-        logger.info('Downloading pair.')
-        exchange.refresh_tickers([pair], tick_interval)
-        tickers[pair] = exchange.klines(pair)
+        logger.info('Downloading pairs.')
+        exchange.refresh_tickers(pairs, tick_interval)
+        for pair in pairs:
+            tickers[pair] = exchange.klines(pair)
     else:
         tickers = history.load_data(
             datadir=Path(_CONF.get("datadir")),
-            pairs=[pair],
+            pairs=pairs,
             ticker_interval=tick_interval,
             refresh_pairs=_CONF.get('refresh_pairs', False),
             timerange=timerange,
             exchange=Exchange(_CONF)
         )
 
-        # No ticker found, or impossible to download
-        if tickers == {}:
-            exit()
+    # No ticker found, impossible to download, len mismatch
+    for pair, data in tickers.copy().items():
+        logger.debug("checking tickers data of pair: %s", pair)
+        logger.debug("data.empty: %s", data.empty)
+        logger.debug("len(data): %s", len(data))
+        if data.empty:
+            del tickers[pair]
+            logger.info(
+                'An issue occured while retreiving datas of %s pair, please retry '
+                'using -l option for live or --refresh-pairs-cached', pair)
+    return tickers
+
 
-    # Get trades already made from the DB
-    trades = load_trades(args, pair, timerange)
+def generate_dataframe(strategy, tickers, pair) -> pd.DataFrame:
+    """
+    Get tickers then Populate strategy indicators and signals, then return the full dataframe
+    :return: the DataFrame of a pair
+    """
 
     dataframes = strategy.tickerdata_to_dataframe(tickers)
-
     dataframe = dataframes[pair]
     dataframe = strategy.advise_buy(dataframe, {'pair': pair})
     dataframe = strategy.advise_sell(dataframe, {'pair': pair})
 
-    if len(dataframe.index) > args.plot_limit:
-        logger.warning('Ticker contained more than %s candles as defined '
-                       'with --plot-limit, clipping.', args.plot_limit)
-    dataframe = dataframe.tail(args.plot_limit)
+    return dataframe
 
-    trades = trades.loc[trades['opents'] >= dataframe.iloc[0]['date']]
-    fig = generate_graph(
-        pair=pair,
-        trades=trades,
-        data=dataframe,
-        args=args
-    )
 
-    plot(fig, filename=str(Path('user_data').joinpath('freqtrade-plot.html')))
+def extract_trades_of_period(dataframe, trades) -> pd.DataFrame:
+    """
+    Compare trades and backtested pair DataFrames to get trades performed on backtested period
+    :return: the DataFrame of a trades of period
+    """
+    trades = trades.loc[trades['opents'] >= dataframe.iloc[0]['date']]
+    return trades
 
 
 def generate_graph(pair, trades: pd.DataFrame, data: pd.DataFrame, args) -> tools.make_subplots:
```

Comment:
I think you may want to remove args from this function and instead pass in `indicators1` `indicators2` ...

otherwise you'll need to fake an args-object (for your **dash**board) to reuse this part